### PR TITLE
Update Unicode's data to version 11.0

### PR DESCRIPTION
**Pull request by data-man**
* added the proc ```swapCase*(c: Rune): Rune```
* added the const ```UnicodeVersion```

Further improvements depends on #7902 status.

### CODE REVIEW

File: lib/pure/unicode.nim [hunk 1]
```diff
@@ -20,9 +22,22 @@ type
...
-proc `<=%`*(a, b: Rune): bool = return int(a) <=% int(b)
+proc `<=%`*(a, b: Rune): bool {.borrow.}
 proc `<%`*(a, b: Rune): bool = return int(a) <% int(b)
-proc `==`*(a, b: Rune): bool = return int(a) == int(b)
+proc `==`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a, b: Rune): bool {.borrow.}
+proc `<=`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a: Rune, b: RuneImpl): bool {.borrow.}
```
Comment (GULPF) [inline lib/pure/unicode.nim]: Do we really need all these overloads? The point of a `distinct` type is to specifically __not__ allow things like this.

File: lib/pure/includes/ucdtables.nim [hunk 2]
```diff
@@ -0,0 +1,7555 @@
...
+  (From: 282, To: 283),
+  (From: 284, To: 285),
+  (From: 286, To: 287),
+  (From: 288, To: 289),
+  (From: 290, To: 291),
+  (From: 292, To: 293),
+  (From: 294, To: 295),
+  (From: 296, To: 297),
+  (From: 298, To: 299),
+  (From: 300, To: 301),
```
Comment (Araq) [inline lib/pure/includes/ucdtables.nim]: Well your script should merge these consequtive ranges...

File: lib/pure/unicode.nim [hunk 3]
```diff
@@ -12,6 +12,8 @@
 {.deadCodeElim: on.}  # dce option deprecated
 
 include "system/inclrtl"
+from algorithm import binarySearch
+from sequtils import mapLiterals
```
Comment (Araq) [inline lib/pure/unicode.nim]: Your script can generate code that doesn't depent on ``mapLiterals``.

File: lib/pure/unicode.nim [hunk 4]
```diff
@@ -20,9 +22,22 @@ type
...
-proc `==`*(a, b: Rune): bool = return int(a) == int(b)
+proc `==`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a, b: Rune): bool {.borrow.}
+proc `<=`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a: Rune, b: RuneImpl): bool {.borrow.}
+proc `<=`*(a: Rune, b: RuneImpl): bool {.borrow.}
+
+proc `<`*(a: RuneImpl, b: Rune): bool {.borrow.}
```
Comment (Araq) [inline lib/pure/unicode.nim]: Don't change the interface of ``unicode.nim``.

File: lib/pure/unicode.nim [hunk 5]
```diff
@@ -321,1064 +336,48 @@ proc runeSubStr*(s: string, pos:int, len:int = int.high): string =
...
-    return Rune(c + toTitleSinglets[p+1] - 500)
-  return Rune(c)
+proc cmpRune(e: CaseEntry, c: Rune): int =
+  if likely(c < e.From):
+    return 1
+  if likely(c > e.From):
+    return -1
+  return 0
+
+include "includes/ucdtables"
```
Comment (Araq) [inline lib/pure/unicode.nim]: Nice, but please also add the tool that you used to generate this file.

**Comment by Araq** [comment]
No feedback, closing.

**Pull request closed without merging**